### DISCUSSION TITLE
런칭데이 전 1차 QA

### DISCUSSION THREAD

**Discussion by Jaymyong66** [woowacourse-teams/2024-code-zap#484] [QA]
# 주변 사람들에게 받은 QA를 아래 코멘트로 달아주세요.

**Comment by Jaymyong66** [comment]
### 소파 QA

-  예를 들어, 생성한 템플릿이 아무 것도 없을 경우, 새 템플릿 추가 버튼을 템플릿 자리에 넣자 ⇒ 한 행 차지(템플릿 2개)
- 파일명이 뭔지 모르겠다
- 확장자에 JS 말고 예시를 넣자

- 파일명 유효성 검사 - 특수기호만 안 들어가게
- 스니펫을 입력해주세요 → 경고창 소스코드로 변경
- 어떤 파일이 잘못 되었는지 표시

- 태그 기능을 전혀 몰랐다
- 템플릿 업로드, 편집 시에도 파일 목록이 보이면 좋겠다 ⇒  selectList 추가
- 스크롤 시, 목록 포커스도 함께 바뀌도록 ⇒ IO 달아야함

- 전체 선택 때, 아직 선택 안했으면 흐리게  ⇒

**Comment by HoeSeong123** [comment]
<img width="914" alt="image" src="https://github.com/user-attachments/assets/046911b3-4632-4301-8144-e24590f8bd41">
아래 동그라미 영역을 클릭해도 코드를 입력하는 커서가 생겼으면 좋겠어요.
현재는 정확하게 1번 줄을 클릭해야 코드를 입력할 수 있습니다.

**Reply by HoeSeong123** [reply]
<img width="1074" alt="image" src="https://github.com/user-attachments/assets/8a112e78-ee49-4f9e-bd7a-c26b441ab455">
화면에 3번째 스니펫이 보여도 오른쪽에 있는 동글배기들은 여전히 1번째를 가리키고 있어요.

**Reply by HoeSeong123** [reply]
<img width="170" alt="스크린샷 2024-08-20 16 17 32" src="https://github.com/user-attachments/assets/ca2c1f4d-4476-4a68-bb15-1f57e6954fa8">
<img width="345" alt="스크린샷 2024-08-20 16 17 50" src="https://github.com/user-attachments/assets/599e089a-71e8-41ef-9b1c-82fb384cda48">

카테고리를 길게 입력하면 이렇게 나옵니다.

**Reply by HoeSeong123** [reply]
<img width="896" alt="스크린샷 2024-08-20 16 20 02" src="https://github.com/user-attachments/assets/ba1eb0f5-784e-42ad-9c02-8d89cd800f45">
제목은 이렇게 나와요

**Comment by kyum-q** [comment]
# 태그 삭제 표시가 점차 작아짐

![image](https://github.com/user-attachments/assets/8a797a87-bbfd-4ba8-9548-aa968f33f2f9)

**Comment by jminkkk** [comment]
제목 짤림????!!!!!!!!!!!!!!!1 예뻐~~~~~~~~~~~~₩
<img width="526" alt="image" src="https://github.com/user-attachments/assets/2a210885-4041-49d4-91f6-77b17baa207f">

**Comment by kyum-q** [comment]
## 태그 hover 작동해서 눌러지는 줄 알았음 

![image](https://github.com/user-attachments/assets/648324ee-ba6e-4334-83c9-4d1f2d0e8d75)

**Comment by Jaymyong66** [comment]
워딩 수정 : 선택 삭제 -> 템플릿 선택

**Reply by Jaymyong66** [reply]
### 아르 QA
- (중요) 검색이 뭐에 대한 검색인지 잘 몰라함
- 카테고리 수정 모달에서 버튼처럼 보이지 않아서 멈칫함
- 검색 결과 없을 때, 검색 결과 없음이 떠야할듯. 현재는 새 템플릿을 수정해보세요가 뜸.

**Comment by kyum-q** [comment]
## 확장자 입력 안해도 저장되니까 에러 안떴으면 좋겠음

그냥 넘기고 저장되길래 저장하고 수정하는데 수정 시에 설명 고치는데 또 에러가 뜸

**Comment by kyum-q** [comment]
## BEBEEEEEBE QA

### 날짜 제대로 안됨

**Comment by kyum-q** [comment]
![image](https://github.com/user-attachments/assets/dd7224c3-1b10-4111-93ee-65ffb7abb1ea)

**Comment by Hain-tain** [comment]
<img width="356" alt="image" src="https://github.com/user-attachments/assets/2c65f04e-b9b5-4cbd-9de9-ad6d95e76070">

- 카테고리 편집 모달에서 카테고리 추가 버튼을 누르면 default로 값이 쓰여져 있는게 불편한 것 같아요! 그냥 placeholder만 있어도 될 것 같습니다~!

**Comment by vi-wolhwa** [comment]
# 월하 QA

### 1. 로그인
- 왜 로그인에 실패하는지 알려줬으면 좋겠다.
- 서버 에러로 로그인이 실패했을 때, 서버 문제인지 계정 문제인지 알 수 없었다.

### 2. 업로드 : 카테고리 힌트
- 힌트가 눈에 잘 띄지 않는다.
- 그리고 옵션의 테두리가 더 연한 느낌? 제목과 겹치니 뭔가 이상하다. 분명 내용과도 겹칠텐데<br>
![image](https://github.com/user-attachments/assets/b8b92d21-87a3-4da1-8123-2a2273b1e6b5)

### 3. 업로드 : 카테고리 생성 시 Loding 힌트
- 로딩 힌트가 번쩍이는게 불편하다. 특수한 환경(3G)을 고려하다가 일반적인 환경의 디자인 퀄리티가 저하될 필요는 없다고 생각한다.<br>
![업로드-카테고리추가](https://github.com/user-attachments/assets/e3fa13e2-fe7f-46ec-8060-5a0ba95714c8)

### 4. 업로드 : 파일명 Blur 시 토스트 메시지
- 입력을 완성하고 제출한 것도 아닌데, Blur 시 토스트 메시지가 뜬다. 특히 빈 칸일 때는 아무것도 하지 않은 사용자에게 경고를 띄우고 있다.
- Blur 대신 저장 버튼을 클릭 했을 때만 토스트메시지로 알려주는 것이 좋아보이며 다른 에러와 일관성이 있다.<br>
![업로드-파일명](https://github.com/user-attachments/assets/ca593cc0-3b26-41bd-8a40-eee6003f07fa)

### 5. 업로드 : 파일명 확장자
- 파일명에 확장자 이름을 강제해야 하는가? 사용자에게 오히려 무의미한 불편함을 주는 느낌이다.

### 6. 업로드 : 타이틀, 설명 Overflow
- 타이틀이 오버플로우 되어도 같은 라인에 계속해서 이어 작성할 수 있다. 이는 좁은 화면에서 전체 파일명을 파악하거나 수정하기 어렵게 만든다. 특히 설명의 경우는 사용자의 입장에서 당연히 개행될 것이라고 예상한다.
- 다음의 Velog와 같이 다음 라인으로 개행하는 방식이 있다. 이 방식은 타이틀 이하 UI의 위치가 변경되지만, 사용해보면 전혀 불편함이나 위화감이 없다.<br>
![image](https://github.com/user-attachments/assets/03697a2e-33dd-4d11-a44b-cac868d1f29e)

### 7. 업로드 : 태그 입력
- 태그 뱃지와 입력을 하나의 라인에서 수행할 수 있도록 하는 것이 더 좋을 것 같다.
- 글을 입력하듯이 자연스럽게 태그를 추가하고, `backspace`를 통해 자연스럽게 태그를 지울 수 있어서 UX 측면에서 유리하다.
- 태그를 잘 못 추가했을 때, 마우스로 손을 이동해야 하는 불편함을 해소할 수 있다. 특히 모바일에서는 태그를 지우려면 키패드 종료, 태그 삭제의 두 단계를 거쳐야 하므로 상당히 불편했다. (모바일에서는 스크롤 키패드에 따라 스크롤이 발생하여 화면이 움직일 수도 있다.)<br>
![image](https://github.com/user-attachments/assets/a6f5d2a1-9689-431a-95b6-2f26be18d480)

### 8. 업로드 : 소스코드 입력
- 소스코드의 첫 줄만 클릭이 가능한 문제

### 9. 카테고리 선택 시 최상단 스크롤
- 카테고리를 잘 못 클릭했을 때, 최상단으로 스크롤 되기 때문에 다시 스크롤 해야 하는 불편함이 있다.
- 사용자의 입장에서 카테고리를 선택했을 때 최상단으로 스크롤 되어야 하는 이유가 있는가? 우측하단 최상단 버튼도 있다.
  - 만약 20개의 템플릿 카드가 일렬로 보인다면 최상단을 보고 싶을 수 있다. 그러나 최상단으로 이동하는 버튼이 멀리 있다. 스크롤 이벤트가 필요한 곳(카테고리)과 가까이 있는 것이 유리하지 않을까?<br>
![카테고리 선택](https://github.com/user-attachments/assets/dbcec3c5-0401-4ac2-a7cd-d3892a5b82fd)

### 10. 태그 선택이 너무 아래 있음
- 태그를 선택했는데 카테고리리는 많고, 필터링된 템플릿은 적어서 템플릿이 안보인다.
- 카테고리를 기본적으로 접어놓고, 마우스 호버 시 펼치는 방법이 있다. 괜찮은 디자인은 생각하고 있는데 시간을 갈아 넣을 자신이 없다. 해줄 사람 구합니다.

### 11. 새 탬플릿 생성 버튼
- 새 템플릿 생성 버튼이 고해상도에서 보니까 뭔가 아쉬워 보인다. 화면의 여백을 가득 채우면 더 예쁠 듯.<br>
![image](https://github.com/user-attachments/assets/ff6b923d-12b8-4069-af66-38bfc5aa390e)<br>
![image](https://github.com/user-attachments/assets/83056d86-dde9-4760-87c9-8854a6f6a73d)

### 12. 카테고리 편집 모달 짤림
- 카테고리의 개수가 많아지니, 카테고리 편집 모달이 짤린다. (스크롤도 안됨)
- 카테고리 개수를 지정해도 이건 불가능..
- 스크롤이 된다고 해도 디자인적으로 별로,,<br>
![image](https://github.com/user-attachments/assets/a728e310-6237-4862-b503-14e57e750357)

### 13. 태그 박스 디자인
- 이전 사진 참고
- 태그 박스의 디자인에서 이질감이 느껴진다. 디자인/구현 해줄 사람 구합니다.

### 14. 모달의 최소 크기
- 모달의 최소 크기가 있는 듯.. 하단에 여백이 생긴다. <br>
![image](https://github.com/user-attachments/assets/71cc1f44-786e-4eac-901b-c73127c4ab05)

**Comment by kyum-q** [comment]
![image](https://github.com/user-attachments/assets/b3b7c8f8-7937-4929-8ebc-9c6daa4013fd)

![image](https://github.com/user-attachments/assets/28f5903a-c4b9-4605-8c54-1db02437b472)

> 수정전 데이터와 비교해서 추가 불가능
